### DISCUSSION TITLE
Crystal trimming

### DISCUSSION THREAD

**Discussion by mlegrand2** [Lora-net/LoRaMac-node#1152] [Q&A]
Hello,
I try to change the values of TXA and TXB_TRIM registers (0x0911 + 0x0912) on a sx1261 to adapt the capacitance of the crystal clock (32MHz).
I've read the discussion "How to trim crystal oscillator of sx1261? #846", but this only concerns continuous wave mode, and the function SetTxContinuous is used only for RF adjstements.

How to trim crystal in normal operation? Where do I have to insert the write to TXA and TXB_TRIM registers?

Thank you for your help.

**Comment by mlegrand2** [comment]
Hello,
I added these lines under each command SX126xSetOperatingMode( MODE_TX ); in sx126x.c :
SX126xWriteRegister( REG_XTA_TRIM, XTA_TRIM_VALUE );
SX126xWriteRegister( REG_XTB_TRIM, XTB_TRIM_VALUE );
I added also the defines, under #define REG_XTA_TRIM 0x0911 in sx126x.h : 
#define REG_XTB_TRIM                                0x0912
#define XTA_TRIM_VALUE                          	0x14
#define XTB_TRIM_VALUE                          	0x14
The default values are 0x12 and reset to 0x12 after each tx mode start.
The values (0x14 in my case) have to be defined to adjust the frequency depending on your crystal. 1 unit adds 0.47pF on each pin of the crystal. So, adding 1 unit on each register means - 100Hz correction on the frequency of the crystal.

See Semtech's AN1200.33 for details.

Should I have to do it also before the RX sequence?

Thank you

**Reply by mluis1** [reply]
You should apply these commands every time you set an operating mode different from `Sleep`.
If I am not mistaken the register values are only reset when radio enters in `Sleep` mode.

The correction if required must be applied to all operating modes except `Sleep` as this trimming influences the radio PLL operation.
